python/lsst/cp/pipe/ptc/cpSolvePtcTask.py

```diff
@@ -453,42 +416,19 @@ def _getInitialGoodPoints(means, variances, maxDeviationPositive, maxDeviationNe
 
         Notes
         -----
-        A linear function has a constant ratio, so find the median
-        value of the ratios, and exclude the points that deviate from
-        that by more than a factor of maxDeviationPositive/negative.
-        Asymmetric deviations are supported as we expect the PTC to
-        turn down as the flux increases, but sometimes it anomalously
-        turns upwards just before turning over, which ruins the fits,
-        so it is wise to be stricter about restricting positive
-        outliers than negative ones.
-
-        Too high and points that are so bad that fit will fail will be
-        included Too low and the non-linear points will be excluded,
-        biasing the NL fit.  This function also masks points after the
-        variance starts decreasing.
+        Eliminate points beyond which the variance decreases
         """
-        assert(len(means) == len(variances))
-        ratios = [b/a for (a, b) in zip(means, variances)]
-        medianRatio = np.nanmedian(ratios)
-        ratioDeviations = [0.0 if a < minMeanRatioTest else (r/medianRatio)-1
-                           for (a, r) in zip(means, ratios)]
-
-        # so that it doesn't matter if the deviation is expressed as
-        # positive or negative
-        maxDeviationPositive = abs(maxDeviationPositive)
-        maxDeviationNegative = -1. * abs(maxDeviationNegative)
-
-        goodPoints = np.array([True if (r < maxDeviationPositive and r > maxDeviationNegative)
-                              else False for r in ratioDeviations])
-
-        # Eliminate points beyond which the variance decreases
-        pivot = np.where(np.array(np.diff(variances)) < 0)[0]
-        if len(pivot) > 0:
+        goodPoints = np.ones_like(means, dtype=bool)
+        pivotList = np.where(np.array(np.diff(variances)) < 0)[0]
+        if len(pivotList) > 0:
             # For small values, sometimes the variance decreases slightly
             # Only look when var > self.config.minVarPivotSearch
-            pivot = [p for p in pivot if variances[p] > minVarPivotSearch]
-            if len(pivot) > 0:
-                pivot = np.min(pivot)
+            pivotList = [p for p in pivotList if variances[p] > minVarPivotSearch]
+            if len(pivotList) > 0:
+                # Require that the decrease in variance happen for two
+                # consecutive signal levels
+                pivotIndex = np.min(np.where(np.diff(pivotList) == 1)[0])
+                pivot = pivotList[pivotIndex]
                 goodPoints[pivot+1:len(goodPoints)] = False
```

Comment:
No need to `len(goodPoints)` here, just make the it `goodPoints[pivot+1:] = False` and you're done.